### ISSUE TITLE
Register filetypes for mmex backup files at system level in Linux

### ISSUE THREAD

**Issue by ovari** [moneymanagerex/moneymanagerex#6026]
https://github.com/moneymanagerex/moneymanagerex/issues/1254

Can backup files have a filetype attached?

![image](https://github.com/moneymanagerex/moneymanagerex/assets/17465872/b062b905-4f59-47d2-b694-2e089d749a9c)

Thank you

**Comment by ovari** [comment]
@joshua-stone for backup files, can:
1. something like `MMEX database backup` filetype be added for files with `*.mmb_*.bak` names?
2. Can a different icon be added for `MMEX database backup`?

Thank you

**Comment by vomikan** [comment]
> 1. Can something like `MMEX database backup` filetype be added for files with `*.mmb_*.bak` names?

This is done to avoid a situation where an inexperienced user opens a backup file and cannot find their data. Absolutely does not require changes.

> 2. Can a different icon be added for `MMEX database backup`?

This is done on the OS side. The application is not responsible for this.

**Comment by ovari** [comment]
Thank you for your response. Unfortunately, don't understand your answers. It seems that the current behavior is correct; hence, closing this issue.

If this issue should not be closed, please re-open.

Thank you

**Comment by tactilis** [comment]
@ovari 
>> Can something like MMEX database backup filetype be added for files with *.mmb_*.bak names?

@vomikan 
> This is done to avoid a situation where an inexperienced user opens a backup file and cannot find their data. Absolutely does not require changes.

I agree.

<br>

@ovari 
> Thank you for your response. Unfortunately, don't understand your answers.

Let me try to explain...
The file extension '.bak' cannot be registered as an MMEX-specific file type because '.bak' (and the fact that it's a SQLite database) may also be used by other applications.

A bit of history, which further explains @vomikan's answer...

MMEX backup files **_used_** to have the extension .mmb
You could tell what the file was only by looking at the file name:

| Original file naming | Purpose |
|----------------------------|---|
|<*database name*>.mmb| MMEX database |
|<*database name*>.mmb\_start_<*yyyy_mm_dd*>.mmb|Backup of MMEX database made at MMEX start|
|<*database name*>.mmb\_update_<*yyyy_mm_dd*>.mmb|Backup of MMEX database made at MMEX exit|
|<*database name*>.mmb\_upgrade_v*NN*_<*yyyy_mm_dd*>.mmb|Backup of MMEX database made during DB upgrade|

Unfortunately, we had many instances where people opened an .mmb **_backup_** file as if it was their latest DB and this led to a proliferation of files named:

<*database name*>.mmb\_update_<*yyyy_mm_dd*>.mmb\_start_<*yyyy_mm_dd*>.mmb
<*database name*>.mmb\_update_<*yyyy_mm_dd*>.mmb\_update_<*yyyy_mm_dd*>.mmb

...sometimes multiple levels deep. Users were understandably confused.

This is why we changed the backup file extension to .bak to avoid the application being able to open these files directly without explicit user action to rename the file to .mmb.

You'll find some discussion of this change here https://github.com/moneymanagerex/moneymanagerex/issues/3648 and info on how to recover a backup file here https://github.com/moneymanagerex/moneymanagerex/issues/4995

**Comment by ovari** [comment]
Thank you for your explanation.

The number of backup files might be quite large over time.

1. Are backup files every automatically deleted?
2. Or is there a prompt to delete backup files?

Thank you

**Comment by tactilis** [comment]
@ovari 
> The number of backup files might be quite large over time.

Please read the Help or see my comment https://github.com/moneymanagerex/moneymanagerex/issues/4995#issuecomment-1230883812 in issue #4995 that I gave you a link to.
